chatcommands.lua

```diff
@@ -7,35 +21,85 @@ minetest.register_chatcommand("channel", {
 	func = function(name, param)
 		if param == "" then
 			minetest.chat_send_player(name, "Online players: /channel online")
-			minetest.chat_send_player(name, "Join/switch:    /channel set <channel>")
+			minetest.chat_send_player(name, "Join/switch:	/channel join <channel>")
 			minetest.chat_send_player(name, "Leave channel:  /channel leave")
+			minetest.chat_send_player(name, "Invite to channel:  /channel invite <playername>")
 			return
+
 		elseif param == "online" then
 			channels.command_online(name)
 			return
+
 		elseif param == "leave" then
 			channels.command_leave(name)
 			return
 		end
+
+
 		local args = param:split(" ")
-		if args[1] == "set" then
-			if #args >= 2 then
-				 channels.command_set(name, args[2])
-				 return
-			end
+
+		if args[1] == "join" and #args == 2 then
+			channels.command_set(name, args[2])
+			return
+
+		elseif args[1] == "invite" and #args == 2 then
+			channels.command_invite(name, args[2])
+			return
+
+		elseif args[1] == "wall" and #args >= 2 then
+			channels.command_wall(name, tablejoin(args,2) )
+			return
 		end
+
 		minetest.chat_send_player(name, "Error: Please check again '/channel' for correct usage.")
 	end,
 })
 
-function channels.say_chat(name, message, channel)
-	for k,v in pairs(channels.players) do
-		if v == channel and k ~= name then
-			minetest.chat_send_player(k, message)
+function channels.say_chat(sendername, message, channel)
+	-- message must already have '<player name>' at start if from a player
+	-- if channel==nil then message is sent only to players in global chat
+
+	minetest.log("action","CHAT: #"..tostring(channel or "no channel").." "..message)
```

Comment:
Superfluous `tostring`. It's always a string. Missing spaces around binary operators:
`minetest.log("action", "CHAT: #" .. (channel or "no channel") .. " " .. message)`

It would be great if you could replace/adjust the other occurrences too. Thanks.